src/GraphQL/Subscription/ObservableExtensions.cs

```diff
@@ -0,0 +1,252 @@
+                // ensure that the transform cannot directly throw an exception without it being wrapped in a Task<TOut>
+                _transformNext = async (data, token) => await transformNext(data, token).ConfigureAwait(false);
+            }
+
+            public void OnNext(TIn value) => Queue(QueueType.Data, _transformNext(value, _token), default);
+
+            public void OnError(Exception error) => Queue(QueueType.Error, default, _transformError(error, _token));
+
+            public void OnCompleted() => Queue(QueueType.Completion, default, default);
+
+            /// <summary>
+            /// Queues the specified event and if necessary starts watching for an event to complete.
+            /// </summary>
+            private void Queue(QueueType queueType, ValueTask<TOut> task, ValueTask<Exception> error)
+            {
+                var queueData = new QueueEvent { Type = queueType, Data = task, Error = error };
+                bool attach = false;
+                lock (_queue)
+                {
+                    _queue.Enqueue(queueData);
+                    attach = _queue.Count == 1;
+                }
+
+                // start watching for an event to complete, if this is the first in the queue
+                if (attach)
+                {
+                    // start sending data events (will await on the task queued if needed)
+                    _ = ProcessAllEventsInQueueAsync();
+                }
+            }
+
+            /// <summary>
+            /// Processes data from the queue in order (or raises errors or completed notifications);
+            /// executes until the queue is empty.
+            /// </summary>
+            private async Task ProcessAllEventsInQueueAsync()
+            {
+                // grab the event at the start of the queue, but don't remove it from the queue
+                QueueEvent queueEvent;
+                bool moreEvents;
+                lock (_queue)
+                {
+                    // should always successfully peek from the queue here
+                    moreEvents = _queue.TryPeek(out queueEvent);
+                }
+                while (moreEvents)
+                {
+                    // process the event
+                    if (queueEvent.Type == QueueType.Data)
+                    {
+                        await ProcessDataAsync(queueEvent.Data).ConfigureAwait(false);
+                    }
+                    else if (queueEvent.Type == QueueType.Error)
+                    {
+                        await ProcessErrorAsync(queueEvent.Error).ConfigureAwait(false);
+                    }
+                    else if (queueEvent.Type == QueueType.Completion)
+                    {
+                        ProcessCompletion();
+                    }
+                    // once the event has been passed along, dequeue it
+                    lock (_queue)
+                    {
```

Comment:
```suggestion
                    {
                        if (_token.IsCancellationRequested)
                        {
                            _queue.Clear();
                            break;
                        }
```

fast return to not process noop possible 10000000 messages in loop after Dispose beeng called.

Reply:
You may emulate and verify this with test with fast source stream and slow response stream.

Reply:
I will make the change you requested; but adding a test for this is not worth it to me.

Reply:
As you see fit.